### ISSUE TITLE
findInCurrentFile stopped working

### ISSUE THREAD

**Issue by 4044ever** [ArturoDent/find-and-transform#8]
I have some find and replace routine set to ALT+Y and ALT+M - today it stopped working.

VS Code
Version: 1.69.1 
Node.js: 16.13.2

Message:
`command 'findInCurrentFile' not found`

I am not sure what happened.

Find and Transform version:
3.3.1 (not working)
3.3.0 (not working)
3.2.6 (**working**!)

My code:

```
[{
    "key": "alt+y",
    "command": "findInCurrentFile",
    "args": {
        "find": ["https://www.youtube",".com/watch\\?v=","(^({{< lite-youtube ).*)"],
        "replace": ["{{< lite-youtube ", "","$1 >}}\n<p>" ],
        "isRegex": true
      }
},{
  "key": "alt+m",
  "command": "findInCurrentFile",
  "args": {
    "find": [
        " für ",  "## Kurzbeschreibung",  "## Wunschinvestment", "## Bekommen", "## Kommentar", "Firmenanteile", "Unternehmensanteile", " von ", "Leider kein Deal" , "/de/loewen/", "Okt.", " Mai "
            ],
 "replace": [
     " for ", "## Description", "## Investment wanted", "## Got a deal?", "## Comment" , "company shares" , "company shares"," from ","Sorry, no deal", "/en/loewen/", "Oct.", " May "
            ],
    "isRegex": false
  }
}]
```

**Comment by Irian-Adappty** [comment]
Precisely today I have updated to VSCode version 1.69.1 and all my other extensions stopped working. There were no crash logs or anything I could check to get me info on what was failing.

So to diagnose the issue I ran 'Start Extension Bisect' command and it also didn't found any culprit. However, when checking the Extensions tab this was the only non Microsoft made extension that wasn't disabled, so Bisect didn't disable it automatically. After uninstalling this extension everything is working again.

Probably this new VSCode version is incopatible with this extension right now.

**Comment by ArturoDent** [comment]
@4044ever It would be helpful if you showed the text you are trying to find and replace.  Nevertheless, I see an issue with your escaping in the first keybinding.  Try this instead:

```jsonc
{
  "key": "alt+y",
  "command": "findInCurrentFile",
  "args": {
    
      "find": ["https://www.youtube","\\.com\\\/watch\\?v=","(^({{< lite-youtube ).*)"],
      
      "replace": ["{{< lite-youtube ", "","$1 >}}\n<p>" ],
      
      "isRegex": true
    }
},
```

You needed more escapes in this replacement: `\\.com\\\/watch\\?v="`

With that change it is working for me, although I don't know the exact text you are working with.  Demo:
![FandTIssue8Demo](https://user-images.githubusercontent.com/31813701/178842347-c259f3d8-19aa-455a-9093-91980efd07be.gif)


If that change to your replace above doesn't fix the issue for you, let me know with examples of your text before and expected text after.

If it is fixed, you can go ahead and close this issue.  Thanks.

**Comment by ArturoDent** [comment]
Also the message `command 'findInCurrentFile' not found` can happen if you try to run the keybindings before the vscode has loaded the extension.  I.e., too soon after opening the window.  You should also test with the updated v3.3.1 of this extension to see if that helped.

**Comment by Zxynine** [comment]
> @Irian-Adappty Please check with the latest version of this extension, v3.3.1. I released a bug yesterday which has already been fixed.
> 
> If the extension doesn't work after updating to v3.3.1, please file a separate issue with the extension keybindings and/or settings you are trying to use and the text/code it should find/transform. I don't believe your issue is related to @4044ever's.

Please be careful with that, I had my entire vscode sort of die on me and I spent half a day trying to figure out what was causing it, I had no errors in any consoles and vs code was frequently trying to start my extensions but nothing would happen, it took me multiple uninstalls to finally see that this extention was stuck in some limbo state i couldnt disable. I wasnt even able to log in to vscode. After reading this I tried just disabling all extensions, uninstalling this one and then enable them one by one to see if I could use them. I think the issue happened when I updated vscode and was testing my extention (I dont shut my computer off frequently so when I updated vscode it probably finalised the update for the broken version you published) I had no idea if it was my fault or not.

**Comment by ArturoDent** [comment]
@Irian-Adappty You say the problems started with the vscode 1.69.1 update.  Were you on v1.69 and using this extension before 1.69.1 and didn't notice any issues?  Only after 1.69.1?

**Comment by 4044ever** [comment]
No matter what, with 3.3.1 I always get `command 'findInCurrentFile' not found`. Seems the extension does not load at all.

The extra escape also works with 3.2.6 - so I keep that anyway.

Since I can use 3.2.6 there is no urgency.

**Comment by ArturoDent** [comment]
@Irian-Adappty Thank you for the additional information.  I found that vscode automatically added some other fields, related to `underDevelopment`, to the `package.json`.  I have seen it once before and it screws up the production version.  I have reverted the package.json to where I think it should be in v3.3.2.

Could I ask you to install v3.3.2 and let me know if it works for you?

Thank you very much for your time on this.  It is working as it should for me all along so it is difficult to figure out what is wrong.

----------

just to document for myself.  Here are the additional fields automatically added to package.json by vscode :

```jsonc
"identifier": {
  "value": "ArturoDent.find-and-transform",
  "_lower": "arturodent.find-and-transform"
 },
 "isBuiltin": false,
 "isUserBuiltin": false,
 "isUnderDevelopment": true,
 "extensionLocation": {
  "$mid": 1,
  "fsPath": "c:\\Users\\Mark\\find-and-transform",
  "_sep": 1,
  "external": "file:///c%3A/Users/Mark/find-and-transform",
  "path": "/c:/Users/Mark/find-and-transform",
  "scheme": "file"
 },
 "targetPlatform": "undefined",
```

possibly a result of testing pre-release versions of the extension and using the released builds on the same instance of vscode?

**Comment by ArturoDent** [comment]
Stack Overflow question: [how to prevent devlopment-related fields automatically added to extension package.json](https://stackoverflow.com/questions/72983445/vscode-extension-how-to-prevent-development-related-fields-automatically-added).

**Comment by Irian-Adappty** [comment]
Yes! It works! Thank you for fixing it your extension is very useful.

I have installed it, closed VSCode and reopen it and everything seems working fine :)

**Comment by ArturoDent** [comment]
@Irian-Adappty Thank you very much for checking the latest update.  

If possible, I would like to hear from @4044ever and @Zxynine about v3.3.2.  Thank you.

**Comment by 4044ever** [comment]
3.3.2 work for me perfectly! Thank you!

**Comment by ArturoDent** [comment]
With no further reports and the known issue of the package.json getting modified by vscode automatically in development having been resolved, I am closing this as completed.

**Comment by ArturoDent** [comment]
I found this github issue, https://github.com/microsoft/vscode/issues/148975, which reported the same behaviour when testing extensions and the `package.json` being silently modified with the same extra metadata which caused those users significant problems.  I am told in that issue that the fix coming in v1.71 should prevent overwriting in the future and the issue hopefully cannot happen again.
